cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRunnerClassLoader.java

```diff
@@ -131,9 +131,13 @@ public SparkRunnerClassLoader(URL[] urls, @Nullable ClassLoader parent, boolean
     // cdap-api-classes
     // Any class that is not from cdap-api-spark or cdap-spark-core
     // Any class that is not from Spark
+    // We also need to define the org.spark-project., fastxml and akka classes in this ClassLoader
+    // to avoid reference/thread leakage due to Spark assumption on process terminating after execution
     if (API_CLASSES.contains(name) || (!name.startsWith("co.cask.cdap.api.spark.")
```

Comment:
It's done by the caller. See https://github.com/caskdata/cdap/pull/5591/commits/e02886cc39bf4660b2790dac193d20b5d0d78c3e